### PR TITLE
[NEW] Option to enable/disable auto away and configure timer

### LINKED ISSUE #736: Option to enable/disable auto away and configure timer
Rocket.Chat keeps user status as `Online` only if user is actively using chat window or if chat window is hovered via mouse. 

Otherwise, user status falls onto `Away`...
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/26534672-set-user-status-correctly?utm_campaign=plugin&utm_content=tracker%2F16890630&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F16890630&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by armand1m**
@RocketChat/core 

Closes #736 

This PR adds two new options to the Account Preferences, where the user can:

 - Enable Auto Away
 - Configure a default Idle Time Limit (only works when auto away is enabled, of course)

This PR also fixes a bug I've found in the `checked` template helper of the account settings package. When values where persisted as false values, but their default ones where `true` values, the `checked` helper would return always the default value since the way it was accessing the nested properties and verifying if they exists where the same as searching for its value.

screenshot_27

### CODE REVIEW

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 1]
```diff
@@ -568,6 +568,8 @@
   "Enable_Svg_Favicon": "Enable SVG favicon",
   "Enable_two-factor_authentication": "Enable two-factor authentication",
   "Enabled": "Enabled",
+  "Enable_Auto_Away": "Enable Auto Away",
+  "Enable_Svg_Favicon": "Enable SVG favicon",
```
Comment (rodrigok) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: Is this string addition correct?

File: tests/pageobjects/side-nav.page.js [hunk 2]
```diff
@@ -39,6 +39,10 @@ class SideNav extends Page {
 	get avatar() { return browser.element('[href="/changeavatar"]'); }
 	get preferencesClose() { return browser.element('.sidebar-flex__close-button[data-action="close"]'); }
 
+	get accountStatusBullet() { return browser.element('.sidebar__account-status-bullet'); }
+	get accountStatusBulletOnline() { return browser.element('.sidebar__account-status-bullet--online'); }
+	get accountStatusBulletAway() { return browser.element('.sidebar__account-status-bullet--away'); }
```
Comment (rodrigok) [inline tests/pageobjects/side-nav.page.js]: Why did you add this?

Comment (armand1m) [reply tests/pageobjects/side-nav.page.js]: Hi, this was meant to be used in e2e tests, but I actually saw that there already were some tests for this functionality. I should have not commited this, I'm removing it.

File: packages/rocketchat-i18n/i18n/en.i18n.json (hunk 1 above)
Comment (armand1m) [reply packages/rocketchat-i18n/i18n/en.i18n.json]: I believe that the "Enable Svg Favicon" came from a merge made to resolve some conflicts. I may need to look at this PR again now :/

**Comment by armand1m** [comment]
Hey, sorry, I've been busy with work. Today I'll review this PR and update it accordingly, as well as fix the broken tests

**Comment by rodrigok** [comment]
@armand1m It's awesome to know that you are working on it again. Let me know if you can fix that things soon, I'm merging PRs to ship a new release candidate tomorrow and I'd love to include your PR.

**Comment by armand1m** [comment]
@rodrigok nice, I'll give it some attention now so we can close this issue. Expect to have this PR prepared today.

**Comment by armand1m** [comment]
@rodrigok Got the tests passing 😄 

I've cleaned the dead code that I've left in the first version. I've also fixed some other tests mistakes.
I've implemented internationalization of this feature in 4 languages:

- English (default)
captura de tela 2017-12-04 as 23 04 00

- Portuguese
captura de tela 2017-12-04 as 23 04 12

- Brazilian Portuguese
captura de tela 2017-12-05 as 00 42 34

- Dutch (Nederlands)
captura de tela 2017-12-05 as 00 42 23

Here is a GIF demonstrating the feature.

enableautoaway

**Comment by armand1m** [comment]
Hope to see it released tomorrow :smile:

File: packages/rocketchat-ui-account/client/accountPreferences.js [hunk 3]
```diff
@@ -152,19 +179,43 @@ Template.accountPreferences.onCreated(function() {
...
-			return s.trim(e);
-		}));
 		data.desktopNotificationDuration = $('input[name=desktopNotificationDuration]').val();
 		data.desktopNotifications = $('#desktopNotifications').find('select').val();
 		data.mobileNotifications = $('#mobileNotifications').find('select').val();
 		data.unreadAlert = $('#unreadAlert').find('input:checked').val();
 		data.notificationsSoundVolume = parseInt($('#notificationsSoundVolume').val());
-
 		data.roomCounterSidebar = $('#roomCounterSidebar').find('input:checked').val();
+		data.messageCounterSidebar = $('#messageCounterSidebar').find('input:checked').val();
```
Comment (rodrigok) [inline packages/rocketchat-ui-account/client/accountPreferences.js]: What is that item? I think it's old code, was replaced by `roomCounterSidebar`

Comment (armand1m) [reply packages/rocketchat-ui-account/client/accountPreferences.js]: Must have been placed in the last merge conflict I needed to solve. Sorry about that.

Comment (rodrigok) [reply packages/rocketchat-ui-account/client/accountPreferences.js]: No problem, it was fixed

**Comment by armand1m** [comment]
Ok, guess it can be merged 😄
